### ISSUE TITLE
[bug] `<rh-pagination>`: manually setting `aria-current="page"` does not work

### ISSUE THREAD

**Issue by adamjohnson** [RedHat-UX/red-hat-design-system#2423]
### Describe the bug

If a user adds `aria-current="page"` to a link in the lightdom, `<rh-pagination>` does not recognize it as the currently selected link.

### Screenshots

![Screenshot of rh-pagination with aria-current but unable to find or display current link](https://github.com/user-attachments/assets/b1b09a72-dd8b-43a6-8256-8fa0e269487f)

### Steps to reproduce

1. Visit the repro on [Codepen](https://codepen.io/adamjohnson/pen/vEOKmJx).
2. Note the HTML in the panel includes `aria-current="page"` on the second link in the lightdom.
3. See how no active link is highlighted and the console logs `Could not determine current link`

### Expected behaviour

Setting `aria-current="page"` in the lightdom should set the currently selected page.

### Element HTML

```HTML
<rh-pagination>
  <ol>
    <li><a href="#page-1">1</a></li>
    <li><a aria-current="page" href="#page-2">2</a></li>
    <li><a href="#page-3">3</a></li>
    <li><a href="#page-4">4</a></li>
    <li><a href="#page-5">5</a></li>
  </ol>
</rh-pagination>
```

### Operating System (OS)

macOS

### Browser

Microsoft Edge

### Additional browser or assistive technology info

Lines 248-251 in rh-pagination.ts query the lightdom for `aria-current`.
